test/integration-test-main.sh

```diff
@@ -717,10 +713,29 @@ function test_multipart_mix {
        return 1
     fi
 
+    # [NOTE]
+    # For macos, in order to free up disk space for statvfs (or df command),
+    # it is necessary to zero out the file size, delete it, and sync it.
+    # In the case of macos, even if you delete a file, there seems to be a
+    # delay in the free space being reflected.
+    # Testing the ensure_diskfree option shows that if this is not done, free
+    # disk space will be exhausted.
+    #
+    if [ "$(uname)" = "Darwin" ]; then
+        cat /dev/null > "${TEMP_DIR}/${BIG_FILE}"
+        cat /dev/null > "${TEMP_DIR}/${BIG_FILE}-mix"
+        cat /dev/null > "${BIG_FILE}"
+        cat /dev/null > "${BIG_FILE}-mix"
```

Comment:
I wonder how this is happening.  Perhaps Time Machine/backup software is running?  I wonder if we could disable it for test directories?

Reply:
I also tried various things.
_(I don't think it's Time Machine. (I think it is possible to check the macos process, but I have not confirmed it))_

Simply deleting test files, including cache files, it seems that is often not enough to recover disk space.

This problem is unlikely to become apparent during normal testing, but it rarely occurred on macos 12.
Then, I tried running `test_multipart_mix` 20-50 times in a row and it was able to confirm that this problem was manifested.
At that time, when I checked the free disk space every time I executed `test_multipart_mix`, I was able to see the increase and decrease.
When the test fails, free disk space has not been recovered.

If s3fs is started in a mode that does NOT use cache, it creates a temporary file(tmpfile), so I think a possible that the space in that temporary file may not be freed.
However, the test that confirmed this error was using a cache file.

I eventually knew that we could restore free disk space by reducing this file size to zero and deleting it and calling sync.
(It's possible that it depended on APFS, but I'm not familiar with that.)